File: Proyecto/Api/WebDev.Api/WebDev.Api/Model/Concept.cs
﻿using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace WebDev.Api.Model
{
    using System.ComponentModel.DataAnnotations;
    public class Concept
    {
        [Key]
        [Required]
        public int Id { get; set; }

        public string Pxordx { get; set; }

        public string Oldpxordx { get; set; }

        public string CodeType { get; set; }

        public string Concept_Class_Id { get; set; }

        [Required]
        public int Concept_Id { get; set; }

        [Required]
        public string Vocabulary_Id { get; set; }

        [Required]
        public string Domain_Id { get; set; }

        public string Track { get; set; }

        public string Standard_Concept { get; set; }

        [Required]
        public string Code { get; set; }

        public string CodeWithPeriods { get; set; }

        public string CodeScheme { get; set; }

        public string Long_Desc { get; set; }

        public string Short_Desc { get; set; }

        public string Code_Status { get; set; }

        public string Code_Change { get; set; }

        public int Code_Change_Year { get; set; }

        public string Code_Planned_Type { get; set; }

        public string Code_Billing_Status { get; set; }

        public string Code_Cms_Claim_Status { get; set; }

        public string Sex_Cd { get; set; }

        public string Anat_Or_Cond { get; set; }

        public string Poa_Code_Status { get; set; }

        public string Poa_Code_Change { get; set; }

        public string Poa_Code_Change_Year { get; set; }

        public string Valid_Start_Date { get; set; }

        public string Valid_End_Date { get; set; }

        public string Invalid_Reason { get; set; }

        public int Create_Dt { get; set; }
    }
}

File: Proyecto/conceptsReact/front-concepts/src/app/models/Concept.js
export default class Concept{
  constructor (id, pxordx, oldpxordx, codeType, concept_Class_Id, concept_Id, vocabulary_Id, domain_Id, track,
    standard_Concept, code, codeWithPeriods, codeScheme, long_Desc, short_Desc, code_Status, code_Change,
    code_Change_Year, code_Planned_Type, code_Billing_Status, code_Cms_Claim_Status, sex_Cd, anat_Or_Cond,
    poa_Code_Status, poa_Code_Change, poa_Code_Change_Year, valid_Start_Date, valid_End_Date, invalid_Reason,
    create_Dt){
    this.id = id; //show
    this.pxordx = pxordx;
    this.oldpxordx = oldpxordx;
    this.codeType = codeType;
    this.concept_Class_Id = concept_Class_Id;
    this.concept_Id = concept_Id; //show
    this.vocabulary_Id = vocabulary_Id;
    this.domain_Id = domain_Id;
    this.track = track;
    this.standard_Concept = standard_Concept;
    this.code = code;
    this.codeWithPeriods = codeWithPeriods;
    this.codeScheme = codeScheme;
    this.long_Desc = long_Desc;
    this.short_Desc = short_Desc; //show
    this.code_Status = code_Status;
    this.code_Change = code_Change;
    this.code_Change_Year = code_Change_Year;
    this.code_Planned_Type = code_Planned_Type;
    this.code_Billing_Status = code_Billing_Status;
    this.code_Cms_Claim_Status = code_Cms_Claim_Status;
    this.sex_Cd = sex_Cd;
    this.anat_Or_Cond = anat_Or_Cond;
    this.poa_Code_Status = poa_Code_Status;
    this.poa_Code_Change = poa_Code_Change;
    this.poa_Code_Change_Year = poa_Code_Change_Year;
    this.valid_Start_Date = valid_Start_Date;
    this.valid_End_Date = valid_End_Date;
    this.invalid_Reason = invalid_Reason;
    this.create_Dt = create_Dt;
  }
}
